Tests/FluentAssertions.Json.Shared.Specs/JTokenAssertionsSpecs.cs

```diff
@@ -65,103 +65,114 @@ public void When_both_objects_are_the_same_or_equal_BeEquivalentTo_should_succee
+                };
+                yield return new object[]
+                {
                     "{ tree: { } }",
                     "{ tree: \"oak\" }",
-                    "has a different type at $.tree")
-                ,
-                Tuple.Create(
+                    "has an object instead of a string at $.tree"
+                };
+                yield return new object[]
+                {
                     "{ tree: { leaves: 10} }",
                     "{ tree: { branches: 5, leaves: 10 } }",
-                    "misses property $.tree.branches")
-                ,
-                Tuple.Create(
+                    "misses property $.tree.branches"
+                };
+                yield return new object[]
+                {
                     "{ tree: { branches: 5, leaves: 10 } }",
                     "{ tree: { leaves: 10} }",
-                    "has extra property $.tree.branches")
-                ,
-                Tuple.Create(
+                    "has extra property $.tree.branches"
+                };
+                yield return new object[]
+                {
                     "{ tree: { leaves: 5 } }",
                     "{ tree: { leaves: 10} }",
-                    "has a different value at $.tree.leaves")
-                ,
-                Tuple.Create(
+                    "has a different value at $.tree.leaves"
+                };
+                yield return new object[]
+                {
                     "{ eyes: \"blue\" }",
                     "{ eyes: [] }",
-                    "has a different type at $.eyes")
-                ,
-                Tuple.Create(
+                    "has a string instead of an array at $.eyes"
+                };
+                yield return new object[]
+                {
                     "{ eyes: \"blue\" }",
                     "{ eyes: 2 }",
-                    "has a different type at $.eyes")
-                ,
-                Tuple.Create(
+                    "has a string instead of an integer at $.eyes"
+                };
+                yield return new object[]
+                {
                     "{ id: 1 }",
                     "{ id: 2 }",
-                    "has a different value at $.id")
-            };
-
-            foreach (var testCase in testCases)
-            {
-                string actualJson = testCase.Item1;
-                string expectedJson = testCase.Item2;
-                string expectedDifference = testCase.Item3;
+                    "has a different value at $.id"
+                };
+            }
+        }
 
-                var actual = (actualJson != null) ? JToken.Parse(actualJson) : null;
-                var expected = (expectedJson != null) ? JToken.Parse(expectedJson) : null;
+        [Theory, MemberData(nameof(FailingBeEquivalentCases))]
+        public void When_objects_differ_BeEquivalentTo_should_fail(string actualJson, string expectedJson,
```

Comment:
Rather then referring to the name of the method, I suggest to rewrite it as `When_json_fragements_are_not_equivalent_it_should_throw`